### PR TITLE
CI AppVeyor

### PR DESCRIPTION
**Pull request by mhujer**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | none
| License       | MIT
| Doc PR        | none

This PR adds a config file for AppVeyor which is a CI similar to travis, except they run the builds on Windows machines. Therefore any possible incompatibilites can be easily discovered (ref #2604)

~~**It is still very WIP, I just want to:
1) prevent possibly duplicate work, if someone starts working on the same thing
2) get feedback as early as possible**~~

You can see the build output here https://ci.appveyor.com/project/mhujer/sylius

To integrate it with this project, it is necessary to login to https://ci.appveyor.com/ using github account and enable the build for this repo.

(Symfony has recently started to use AppVeyor - https://github.com/symfony/symfony/pull/15575, that's where most of the config comes from)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     ],
     "require": {
         "php":                                  ">=5.5.9",
+        "ext-exif":                             "*",
```
Comment (mhujer) [inline composer.json]: Exif extension is not required by any package, but if it is missing, it causes error during `cache:warmup`

```
php app\console cache:warmup --env=test

Warming up the cache for the test environment with debug true
                                                                
  [Imagine\Exception\NotSupportedException]                     
  PHP exif extension is required to use the ExifMetadataReader  
                                                                
 
 ```

**Comment by pjedrzejewski** [comment]
This is awesome, thank you for your work Martin! :+1: I hope this is the first step in much better Windows support.

**Comment by adamelso** [comment]
aww beat me to it https://github.com/adamelso/Sylius/pull/6

but nice! :+1:  I'll let you continue

**Comment by pjedrzejewski** [comment]
@adamelso I like to have such problems in the community! :) Let's combine your work and get the best possible configuration. Would be great if you guys discuss your approaches.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - php composer.phar install --prefer-dist --no-interaction --no-progress --ansi
+
+test_script:
+    - php app\console doctrine:database:create --env=test
+    - php app\console doctrine:schema:create --env=test
+    - php app\console doctrine:phpcr:repository:init --env=test
+    - php app\console cache:warmup --env=test
+
+    - IF %TEST_SUITE%==phpspec travis\appveyor\suite-phpspec.cmd
+    - IF %TEST_SUITE%==1a travis\appveyor\suite-1a.cmd
```
Comment (mhujer) [inline appveyor.yml]: I needed to split the existing suites into two parts to fit into AppVeyor build time limit

File: travis/appveyor/suite-phpspec.cmd [hunk 3]
```diff
@@ -0,0 +1,3 @@
+call php app/console sylius:install --no-interaction -vvv
```
Comment (mhujer) [inline travis/appveyor/suite-phpspec.cmd]: This should verify, that Sylius is actually able to install itself

**Comment by pjedrzejewski** [comment]
Great work @mhujer! What configuration should I setup in AppVeyor? I have registered and added Sylius project. I merge this to see if it works, but I guess I will need to do a bit more setup. I really appreciate the effort you have put into this, it will make our life so much easier and hopefully we see less bug reports in Windows. :) :+1:

**Comment by mhujer** [comment]
@pjedrzejewski There shouldn't be any setup necesarry in the AppVeyor - you can either set it through the web UI or through the config file.

**Comment by mhujer** [comment]
The only downside of the AppVeyor is that the build is quite slow :snail:

**Pull request merged**